sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1902,202 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+def _get_cache_dict():
```

Comment:
I don't really like this part, but I am not really sure how else to do it, because if I just do defaultdict(lambda: defaultdict(int)), the thing will not pickle.